core/src/main/java/com/scalar/db/transaction/consensuscommit/CommitHandler.java

```diff
@@ -85,21 +97,12 @@ private void prepareRecords(Snapshot snapshot)
     parallelExecutor.prepare(tasks, snapshot.getId());
   }
 
-  public void preCommitValidation(Snapshot snapshot, boolean abortIfError)
-      throws CommitException, UnknownTransactionStatusException {
-    // pre-commit validation is executed when SERIALIZABLE with EXTRA_READ strategy is chosen.
+  public void validate(Snapshot snapshot) throws ValidationException {
```

Comment:
Renamed `preCommitValidation()` to `validate()`.

And I made this method throw `ValidationException` instead of `CommitException`.